src/grep.rs

```diff
@@ -152,8 +159,7 @@ fn grep_offset(
     }
   };
 
-
-  let mut matches = pattern.find_iter(buffer);
+  let matches = &mut pattern.find_iter(buffer);
```

Comment:
I believe a reference has no benefit here. If I'm right, could you revert this please?